src/benchmarks/micro/libraries/System.IO.FileSystem/Perf.FileStream.cs

```diff
@@ -2,152 +2,299 @@
+        [Arguments(OneMegaByte, HalfKiloByte, FileOptions.None)]
+        [Arguments(OneMegaByte, FourKiloBytes, FileOptions.None)]
+        [Arguments(HundredMegaBytes, HalfKiloByte, FileOptions.None)]
+        [Arguments(HundredMegaBytes, FourKiloBytes, FileOptions.None)]
+        public void Write(long fileSize, int userBufferSize, FileOptions options)
         {
-            using (var reader = new FileStream(_filePath, FileMode.Open, FileAccess.Read, FileShare.Read, DefaultBufferSize, FileOptions.Asynchronous))
+            byte[] userBuffer = _userBuffers[userBufferSize];
+            using (FileStream fileStream = new FileStream(_destinationFilePaths[fileSize], FileMode.Create, FileAccess.Write, FileShare.Read, FourKiloBytes, options))
             {
-                await reader.CopyToAsync(Stream.Null);
+                for (int i = 0; i < fileSize / userBufferSize; i++)
+                {
+                    fileStream.Write(userBuffer, 0, userBuffer.Length);
+                }
             }
         }
 
         [Benchmark]
-        public void WriteByte()
+        [Arguments(OneKiloByte, HalfKiloByte, FileOptions.Asynchronous)]
+        [Arguments(OneKiloByte, FourKiloBytes, FileOptions.Asynchronous)]
+        [Arguments(OneMegaByte, HalfKiloByte, FileOptions.Asynchronous)]
+        [Arguments(OneMegaByte, FourKiloBytes, FileOptions.Asynchronous)]
+        [Arguments(HundredMegaBytes, HalfKiloByte, FileOptions.Asynchronous)]
+        [Arguments(HundredMegaBytes, FourKiloBytes, FileOptions.Asynchronous)]
+        [BenchmarkCategory(Categories.NoWASM)]
+        public async Task<long> ReadAsync(long fileSize, int userBufferSize, FileOptions options)
         {
-            using (FileStream writer = new FileStream(_filePath, FileMode.Create, FileAccess.Write, FileShare.Read, DefaultBufferSize, FileOptions.None))
+            byte[] userBuffer = _userBuffers[userBufferSize];
+            long bytesRead = 0;
+            using (FileStream fileStream = new FileStream(_sourceFilePaths[fileSize], FileMode.Open, FileAccess.Read, FileShare.Read, FourKiloBytes, options))
             {
-                for (int i = 0; i < TotalSize; i++)
+                while (bytesRead < fileSize)
                 {
-                    writer.WriteByte(default);
+                    bytesRead += await fileStream.ReadAsync(userBuffer, 0, userBuffer.Length);
                 }
             }
+
+            return bytesRead;
         }
-        
+
         [Benchmark]
-        public void Write()
+        [Arguments(OneKiloByte, HalfKiloByte, FileOptions.Asynchronous)]
+        [Arguments(OneKiloByte, FourKiloBytes, FileOptions.Asynchronous)]
+        [Arguments(OneMegaByte, HalfKiloByte, FileOptions.Asynchronous)]
+        [Arguments(OneMegaByte, FourKiloBytes, FileOptions.Asynchronous)]
+        [Arguments(HundredMegaBytes, HalfKiloByte, FileOptions.Asynchronous)]
+        [Arguments(HundredMegaBytes, FourKiloBytes, FileOptions.Asynchronous)]
```

Comment:
Same question.  I'd have thought we'd want FileOptions.None as well.

Reply:
I thought that `FileOptions.None` combined with async file operations is more like an anti-pattern. Is it not?

Reply:
> is more like an anti-pattern

It's not as simple as that.  It's very common to have a file opened for non-async I/O and then use the async methods with it, often because of separation of concerns or composition.  For example, some component opens a file and hands the stream to you, and then you use it as part of an operation like `networkStream.CopyToAsync(destination)`.

Reply:
@stephentoub now I get it. I've added `FileOptions.None` to all async benchmarks. Thanks!